src/daft-sql/src/python.rs

```diff
@@ -58,10 +58,13 @@ pub fn sql_exec(
 pub fn sql(
     sql: &str,
     catalog: PyCatalog,
+    py_session: &PySession,
     daft_planning_config: PyDaftPlanningConfig,
 ) -> PyResult<PyLogicalPlanBuilder> {
     // TODO deprecated catalog APIs #3819
-    let session = Session::empty();
+
+    let session = py_session.inner().fork();
```

Comment:
Not at a blocker, but it might be worth considering a scoped set of bindings (tables and functions) that gets used in the planner context rather than being placed in the session.

This may be simpler and avoids forking the session for every call to SQL. Honestly may overall simplify things, just need to update the `resolve_table` and `resolve_function` methods in the planner. Then there are zero changes to session here and you don't have to pass the py_session or add any of these clone/fork methods.

Reply:
I agree that the forking is not the most optimal approach. What I want to do is refactor the SQLContext to take in a reference to a `ContextProvider` trait, and then we can impl that for PySession and Session, removing the `.session()` method, and the `.fork` all together. Since that's a pretty big refactor, I'd prefer to get this in as is, then clean up the SQLcontext in a subsequent PR.